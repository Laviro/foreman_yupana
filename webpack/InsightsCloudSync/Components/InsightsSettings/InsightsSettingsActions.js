import API from 'foremanReact/API';
import { insightsCloudUrl } from '../../InsightsCloudSyncHelpers';
import {
  INSIGHTS_SYNC_SETTING_SET,
  INSIGHTS_SYNC_SETTING_SET_FAILURE,
  INSIGHTS_SYNC_SETTINGS_GET_SUCCESS,
  INSIGHTS_SYNC_SETTINGS_GET_FAILURE,
} from './InsightsSettingsConstants';

export const getInsightsSyncSettings = () => async dispatch => {
  try {
    const {
      data: { insightsSyncEnabled },
    } = await API.get(insightsCloudUrl('settings'));
    dispatch({
      type: INSIGHTS_SYNC_SETTINGS_GET_SUCCESS,
      payload: {
        settings: {
          insightsSyncEnabled,
        },
      },
    });
  } catch (error) {
    dispatch({
      type: INSIGHTS_SYNC_SETTINGS_GET_FAILURE,
      payload: {
        settings: {
          error: error.message,
        },
      },
    });
  }
};

export const setInsightsSyncEnabled = currentInsightsSyncEnabled => async dispatch => {
  try {
    const {
      data: { insightsSyncEnabled },
    } = await API.patch(insightsCloudUrl('settings'), {
      insightsSyncEnabled: currentInsightsSyncEnabled,
    });
    dispatch({
      type: INSIGHTS_SYNC_SETTING_SET,
      payload: {
        settings: {
          insightsSyncEnabled,
        },
      },
    });
  } catch (error) {
    dispatch({
      type: INSIGHTS_SYNC_SETTING_SET_FAILURE,
      payload: {
        settings: {
          error: error.message,
        },
      },
    });
  }
};
