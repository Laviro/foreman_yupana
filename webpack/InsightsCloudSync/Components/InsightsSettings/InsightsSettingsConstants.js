export const INSIGHTS_SYNC_SETTING_SET = 'INSIGHTS_SYNC_SETTING_SET';

export const INSIGHTS_SYNC_SETTINGS_GET_SUCCESS =
  'INSIGHTS_SYNC_SETTINGS_GET_SUCCESS';
