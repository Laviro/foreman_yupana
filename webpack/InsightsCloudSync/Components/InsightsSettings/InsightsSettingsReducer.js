import Immutable from 'seamless-immutable';
import {
  INSIGHTS_SYNC_SETTING_SET,
  INSIGHTS_SYNC_SETTINGS_GET_SUCCESS,
} from './InsightsSettingsConstants';

const initialState = Immutable({
  insightsSyncEnabled: false,
});

export default (state = initialState, action) => {
  const { payload: { settings } = {} } = action;

  switch (action.type) {
    case INSIGHTS_SYNC_SETTINGS_GET_SUCCESS:
    case INSIGHTS_SYNC_SETTING_SET:
      return state.merge({
        ...state,
        ...settings,
      });
    default:
      return state;
  }
};
