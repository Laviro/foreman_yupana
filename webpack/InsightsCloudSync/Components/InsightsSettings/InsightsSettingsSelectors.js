import { selectInsightsCloudSync } from '../../../ForemanRhCloudSelectors';

export const selectSettings = state => selectInsightsCloudSync(state).settings;

export const selectInsightsSyncEnabled = state =>
  selectSettings(state).insightsSyncEnabled;
