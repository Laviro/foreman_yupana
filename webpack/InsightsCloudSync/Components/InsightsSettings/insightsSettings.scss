.insights_settings {
  position: absolute;
  top: 84px;
  right: 24px;

  .pf-c-switch__label {
    padding-left: 0;
  }

  .switcher-help-label {
    padding-left: 5px;
  }
}
