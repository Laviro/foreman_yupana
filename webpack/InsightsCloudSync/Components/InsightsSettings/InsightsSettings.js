import React, { Component } from 'react';
import PropTypes from 'prop-types';
import InsightsSyncSwitcher from '../InsightsSyncSwitcher';
import './insightsSettings.scss';

class InsightsSettings extends Component {
  componentDidMount() {
    const { getInsightsSyncSettings } = this.props;
    getInsightsSyncSettings();
  }

  render() {
    const { insightsSyncEnabled, setInsightsSyncEnabled } = this.props;
    return (
      <div className="insights_settings">
        <InsightsSyncSwitcher
          insightsSyncEnabled={insightsSyncEnabled}
          setInsightsSyncEnabled={setInsightsSyncEnabled}
        />
      </div>
    );
  }
}

InsightsSettings.propTypes = {
  insightsSyncEnabled: PropTypes.bool.isRequired,
  getInsightsSyncSettings: PropTypes.func.isRequired,
  setInsightsSyncEnabled: PropTypes.func.isRequired,
};

export default InsightsSettings;
