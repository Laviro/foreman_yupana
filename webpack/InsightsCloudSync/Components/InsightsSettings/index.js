import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import InsightsSettings from './InsightsSettings';
import { selectInsightsSyncEnabled } from './InsightsSettingsSelectors';
import * as actions from './InsightsSettingsActions';
import reducer from './InsightsSettingsReducer';

// map state to props
const mapStateToProps = state => ({
  insightsSyncEnabled: selectInsightsSyncEnabled(state),
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export connected component
export default connect(mapStateToProps, mapDispatchToProps)(InsightsSettings);

// export reducers
export const reducers = { settings: reducer };
