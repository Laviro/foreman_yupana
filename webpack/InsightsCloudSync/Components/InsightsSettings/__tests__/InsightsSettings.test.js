import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { noop } from 'patternfly-react';

import InsightsSettings from '../InsightsSettings';

const fixtures = {
  'render without Props': {
    insightsSyncEnabled: false,
    getInsightsSyncSettings: noop,
    setInsightsSyncEnabled: noop,
  },
  /** fixtures, props for the component */
};

describe('InsightsSettings', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(InsightsSettings, fixtures));
});
