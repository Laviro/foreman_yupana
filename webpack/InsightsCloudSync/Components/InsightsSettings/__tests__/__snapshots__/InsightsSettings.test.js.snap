// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsSettings rendering render without Props 1`] = `
<div
  className="insights_settings"
>
  <InsightsSyncSwitcher
    insightsSyncEnabled={false}
    setInsightsSyncEnabled={[Function]}
  />
</div>
`;
